src/unexploitable_search/model_organisms/train_utils.py

```diff
@@ -0,0 +1,59 @@
+from typing import Callable, Any
+import asyncio
+from unexploitable_search.quest import Quest
+from unexploitable_search.settings.apps.utils import preprocess_question
+from unexploitable_search.setting import Setting
+from unexploitable_search.settings.apps.setting import APPSSetting
+from unexploitable_search.settings.alpaca.setting import AlpacaSetting
+from unexploitable_search.settings.apps.quests.side_quests.side_quest import (
+    APPSSideQuest,
+)
+from unexploitable_search.settings.alpaca.quests.side_quests.side_quest import (
+    AlpacaSideQuest,
+)
+
+
+def make_quest_grpo_reward(quest: Quest) -> Callable:  # type: ignore
+    """
+    The Quest.evaluate method requires only a completion, so cannot be used
+    out-of-the-box as a reward for GRPO, because it expects a prompt and completion
+    (even though the prompt is unnecessary for side quest rewards).
+    """
+
+    def reward(
+        prompts: list[str],
+        completions: list[str],
+        **kwargs: Any,
+    ) -> list[float]:
+        async def run():
+            results = await asyncio.gather(*(quest.evaluate(c) for c in completions))
+            return [float(s) for s, _ in results]
+
+        return asyncio.run(run())
+
+    return reward
+
+
+def format_prompt(
+    x: dict[str, Any],
+    setting: Setting,
+    side_quest: AlpacaSideQuest | APPSSideQuest | None = None,
+) -> dict[str, Any]:
+    sys_prompt = "You are a helpful assistant. "
+    if side_quest is not None:
+        sys_prompt += side_quest.prompt
```

Comment:
The side quest prompt should be added in the same way we do for evaluations. At least in APPS case this means a different system prompt with formatting applied. Maybe the setting should be in charge of formatting the prompt